### PR TITLE
Remove usage of TypeSwitch.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/EditorFeatures/Core/Implementation/IntelliSense/Completion/FileSystem/FileSystemCompletionHelper.cs [hunk 1]
```diff
@@ -228,9 +228,13 @@ private bool ShouldShow(FileSystemInfo child)
 
         private bool CanAccess(FileSystemInfo info)
         {
-            return info.TypeSwitch(
-                (DirectoryInfo d) => CanAccessDirectory(d),
-                (FileInfo f) => CanAccessFile(f));
+            switch (info)
+            {
+                case DirectoryInfo d: return CanAccessDirectory(d);
```
Comment (agocke) [inline src/EditorFeatures/Core/Implementation/IntelliSense/Completion/FileSystem/FileSystemCompletionHelper.cs]: Single line statements are prohibited in our style guide.

Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Implementation/IntelliSense/Completion/FileSystem/FileSystemCompletionHelper.cs]: I welcome some sort of formatter that enforces those rules.  The IDE is certainly full of single line switch statements :)

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

File: src/EditorFeatures/CSharp/AutomaticCompletion/AutomaticLineEnderCommandHandler.cs [hunk 2]
```diff
@@ -136,12 +136,15 @@ protected override string GetEndingString(Document document, int position, Cance
                 var textToParse = owningNode.NormalizeWhitespace().ToFullString() + semicolon;
 
                 // currently, Parsing a field is not supported. as a workaround, wrap the field in a type and parse
-                var node = owningNode.TypeSwitch(
-                    (BaseFieldDeclarationSyntax n) => SyntaxFactory.ParseCompilationUnit(WrapInType(textToParse), options: (CSharpParseOptions)tree.Options),
-                    (BaseMethodDeclarationSyntax n) => SyntaxFactory.ParseCompilationUnit(WrapInType(textToParse), options: (CSharpParseOptions)tree.Options),
-                    (BasePropertyDeclarationSyntax n) => SyntaxFactory.ParseCompilationUnit(WrapInType(textToParse), options: (CSharpParseOptions)tree.Options),
-                    (StatementSyntax n) => SyntaxFactory.ParseStatement(textToParse, options: (CSharpParseOptions)tree.Options),
-                    (UsingDirectiveSyntax n) => (SyntaxNode)SyntaxFactory.ParseCompilationUnit(textToParse, options: (CSharpParseOptions)tree.Options));
+                var node = (SyntaxNode)null;
```
Comment (jasonmalinowski) [inline src/EditorFeatures/CSharp/AutomaticCompletion/AutomaticLineEnderCommandHandler.cs]: Eww that's really ugly.

File: src/EditorFeatures/CSharp/Highlighting/KeywordHighlighters/ReturnStatementHighlighter.cs [hunk 3]
```diff
@@ -39,14 +39,13 @@ internal class ReturnStatementHighlighter : AbstractKeywordHighlighter<ReturnSta
...
+            switch (node)
+            {
+                case ReturnStatementSyntax statement:
                     spans.Add(statement.ReturnKeyword.Span);
                     spans.Add(EmptySpan(statement.SemicolonToken.Span.End));
-                },
-                _ =>
-                {
+                    break;
+                default:
```
Comment (jasonmalinowski) [inline src/EditorFeatures/CSharp/Highlighting/KeywordHighlighters/ReturnStatementHighlighter.cs]: Bad spacing.

File: src/EditorFeatures/CSharp/Highlighting/KeywordHighlighters/YieldStatementHighlighter.cs [hunk 4]
```diff
@@ -39,18 +39,17 @@ internal class YieldStatementHighlighter : AbstractKeywordHighlighter<YieldState
...
                         TextSpan.FromBounds(
                             statement.YieldKeyword.SpanStart,
                             statement.ReturnOrBreakKeyword.Span.End));
 
                     spans.Add(EmptySpan(statement.SemicolonToken.Span.End));
-                },
-                _ =>
-                {
+                    break;
+                default:
```
Comment (jasonmalinowski) [inline src/EditorFeatures/CSharp/Highlighting/KeywordHighlighters/YieldStatementHighlighter.cs]: Bad spacing.

File: src/Features/Core/Portable/Shared/Extensions/ISymbolExtensions_2.cs [hunk 5]
```diff
@@ -170,12 +170,15 @@ public static Glyph GetGlyph(this ISymbol symbol)
 
         public static IEnumerable<TaggedText> GetDocumentationParts(this ISymbol symbol, SemanticModel semanticModel, int position, IDocumentationCommentFormattingService formatter, CancellationToken cancellationToken)
         {
-            var documentation = symbol.TypeSwitch(
-                    (IParameterSymbol parameter) => parameter.ContainingSymbol.OriginalDefinition.GetDocumentationComment(cancellationToken: cancellationToken).GetParameterText(symbol.Name),
-                    (ITypeParameterSymbol typeParam) => typeParam.ContainingSymbol.GetDocumentationComment(cancellationToken: cancellationToken).GetTypeParameterText(symbol.Name),
-                    (IMethodSymbol method) => GetMethodDocumentation(method),
-                    (IAliasSymbol alias) => alias.Target.GetDocumentationComment(cancellationToken: cancellationToken).SummaryText,
-                    _ => symbol.GetDocumentationComment(cancellationToken: cancellationToken).SummaryText);
+            var documentation = (string)null;
```
Comment (jasonmalinowski) [inline src/Features/Core/Portable/Shared/Extensions/ISymbolExtensions_2.cs]: Ick.

File: src/Workspaces/CSharp/Portable/Extensions/SyntaxNodeExtensions.cs [hunk 6]
```diff
@@ -826,42 +826,46 @@ public static SyntaxNode GetParent(this SyntaxNode node)
...
+                case UsingStatementSyntax n: return ValueTuple.Create(n.OpenParenToken, n.CloseParenToken);
+                case FixedStatementSyntax n: return ValueTuple.Create(n.OpenParenToken, n.CloseParenToken);
+                case LockStatementSyntax n: return ValueTuple.Create(n.OpenParenToken, n.CloseParenToken);
+                case IfStatementSyntax n: return ValueTuple.Create(n.OpenParenToken, n.CloseParenToken);
+                case SwitchStatementSyntax n: return ValueTuple.Create(n.OpenParenToken, n.CloseParenToken);
+                case CatchDeclarationSyntax n: return ValueTuple.Create(n.OpenParenToken, n.CloseParenToken);
+                case AttributeArgumentListSyntax n: return ValueTuple.Create(n.OpenParenToken, n.CloseParenToken);
+                case ConstructorConstraintSyntax n: return ValueTuple.Create(n.OpenParenToken, n.CloseParenToken);
+                case ParameterListSyntax n: return ValueTuple.Create(n.OpenParenToken, n.CloseParenToken);
+                default: return default(ValueTuple<SyntaxToken, SyntaxToken>);
```
Comment (jasonmalinowski) [inline src/Workspaces/CSharp/Portable/Extensions/SyntaxNodeExtensions.cs]: Not related to you but this is icky. :-(

**Pull request merged**